skema/text_reading/text_reading/grounding/src/main/scala/org/ml4ai/grounding/HyperParameterSearch.scala

```diff
@@ -0,0 +1,90 @@
+  * alpha * cosineSimilarities + (1-alpha) (1-normalized edit distances)
+  */
+object HyperParameterSearch extends App {
+
+
+  /***
+    * Get accuracy for this lambda, alpha values provided.
+    * MiraEmbeddingsGrounder is initialized for lambda, alpha value.
+    * @param lambda
+    * @param alpha
+    * @return accuracy
+    */
+  def getAccuracyForThisHyperParams(lambda: Float, alpha: Float, path: String): Float = {
+    val miraEmbeddingsGrounderGS: MiraEmbeddingsGrounder = {
+      val config = ConfigFactory.load().getConfig("Grounding")
+      val ontologyPath = config.getString("ontologyPath")
+      // val embeddingsPath = config.getString("embeddingsPath")
+      MiraEmbeddingsGrounder(ontologyPath, None, lambda, alpha)
+    }
+
+
+    val targets = {
+      // Drop the first line that is the header
+      FileUtils.getTextFromResource(path = path).split("\n").drop(1) map { line =>
+        val tokens = line.split("\t")
+        (tokens(0), tokens(1))
+      }
+    }
+    print(targets)
+
+    val predictions = {
+      for {(text, groundingId) <- targets
+           } yield miraEmbeddingsGrounderGS.ground(text) match {
+        case Some(concept) => concept.id == groundingId
+        case None => false
+      }
+    }
+    print(predictions)
+    val accuracy = (predictions.count(identity).floatValue() / predictions.length).floatValue()
+    val acc = accuracy.toFloat
+    acc
+  }
+
+  def apply(resourcePath: String, isSimulation:Boolean): ListMap[(Float, Float), Float] = {
+
+    // Grid search -bruteforce
+    var hyperparam_size: Int = 2;
+    var path: String = resourcePath
+    var lambdas = DenseVector[Float](1f, 10f, 100f) //Uniform(10, 1000).samplesVector(hyperparam_size)
+    var alphas = DenseVector[Float](0.25f, 0.5f, 0.75f) //Gaussian(0.0.toFloat, 1.0.toFloat).samplesVector(hyperparam_size).toDenseVector
+    if (isSimulation){
+      lambdas = Uniform(10f, 1000f).samplesVector(hyperparam_size).values.map(_.toFloat).toDenseVector
+      alphas = Gaussian(0.0f, 1.0f).samplesVector(hyperparam_size).values.map(_.toFloat).toDenseVector
+    }
+    //  val lambdas = DenseVector(1, 3, 5, 7, 10, 50, 75, 100) //Uniform(10, 1000).samplesVector(hyperparam_size)
+    //  val alphas = DenseVector(0.25, 0.5, 0.75) //Gaussian(0.0.toFloat, 1.0.toFloat).samplesVector(hyperparam_size).toDenseVector
+
+    val acc_map = new scala.collection.mutable.HashMap[(Float, Float), Float]()
+
+    // for each lambda, alpha value chosen just for a base case, loop mover to calculate accuracy
+    for (lambda <- lambdas.toScalaVector()) yield {
+      for (alpha <- alphas.toScalaVector()) yield {
+        val this_acc = getAccuracyForThisHyperParams(lambda.toFloat, alpha.toFloat, path)
+        acc_map.update((lambda, alpha), this_acc)
+      }
+    }
+    val sorted_accuracy_map = ListMap(acc_map.toSeq.sortWith(_._2 > _._2):_*)
+    println(acc_map)
+    sorted_accuracy_map
+  }
+
+}
+
```

Comment:
@enoriega  I updated and uploaded. Tests on the CI seems to have passed. But, when I ran this test app starts but the path of the grounding tests need to be passed to the app and that path looks like :   
sbt "runMain org.ml4ai.grounding.HyperParameterSearch  this/local/path/to/grounding_tests.tsv" becuase right now this file is only strictly for tests so it is available only within test scope. so how do we add the file and test this locally such that FileUtils.getTextFromResource(path) will not fail?

Reply:
If the file is small enough for github, it should probably be put into skema/text_reading/text_reading/grounding/src/main/resources/somewhere and then it might be referred to as a resource as /somewhere.

Reply:
@kwalcock so another question as a follow up to your response is: the file I am referring to, is used both in TestMiraEmbeddingsGrounder.scala and HyperParameterSearch.scala. So right now, the file is already retrieved and is available at runtime @ grounding/target/scala-2.12/test-classes/grounding_tests.tsv . but I am using the same file to test HyperParameterSearch.scala . Is there another path we can agree on so the same file "grounding_tests.tsv" can be commonly used for testing as well as by other main classes/standalone scripts ?

Reply:
I like @kwalcock 's proposed solution. @sushmaakoju please move `HyperParameterSearch.scala` to the test namespace. Then you can read the file as in the test cases directly.